addons/stock/inherit_stock_picking.py

```diff
@@ -0,0 +1,336 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import date, datetime
+from dateutil import relativedelta
+import json
+import time
+import sets
+
+import openerp
+from openerp.osv import fields, osv
+from openerp.tools.float_utils import float_compare, float_round
+from openerp.tools.translate import _
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT, DEFAULT_SERVER_DATE_FORMAT
+from openerp import SUPERUSER_ID, api, models
+import openerp.addons.decimal_precision as dp
+from openerp.addons.procurement import procurement
+import logging
+from openerp.exceptions import UserError
+
+class stock_picking(models.Model):
+    _inherit = "stock.picking"
+
+    # def create(self, cr, user, vals, context=None):
+    #     context = context or {}
+    #     if ('name' not in vals) or (vals.get('name') in ('/', False)):
+    #         ptype_id = vals.get('picking_type_id', context.get('default_picking_type_id', False))
+    #         sequence_id = self.pool.get('stock.picking.type').browse(cr, user, ptype_id, context=context).sequence_id.id
+    #         vals['name'] = self.pool.get('ir.sequence').next_by_id(cr, user, sequence_id, context=context)
+    #     # As the on_change in one2many list is WIP, we will overwrite the locations on the stock moves here
+    #     # As it is a create the format will be a list of (0, 0, dict)
+    #     if vals.get('move_lines') and vals.get('location_id') and vals.get('location_dest_id'):
+    #         for move in vals['move_lines']:
+    #             if len(move) == 3:
+    #                 move[2]['location_id'] = vals['location_id']
+    #                 move[2]['location_dest_id'] = vals['location_dest_id']
+    #     return super(stock_picking, self).create(cr, user, vals, context)
+
+    # def write(self, cr, uid, ids, vals, context=None):
+    #     res = super(stock_picking, self).write(cr, uid, ids, vals, context=context)
+    #     after_vals = {}
+    #     if vals.get('location_id'):
+    #         after_vals['location_id'] = vals['location_id']
+    #     if vals.get('location_dest_id'):
+    #         after_vals['location_dest_id'] = vals['location_dest_id']
+    #     # Change locations of moves if those of the picking change
+    #     if after_vals:
+    #         moves = []
+    #         for pick in self.browse(cr, uid, ids, context=context):
+    #             moves += [x.id for x in pick.move_lines if not x.scrapped]
+    #         if moves:
+    #             self.pool['stock.move'].write(cr, uid, moves, after_vals, context=context)
+    #     return res
```

Comment:
If no use of this code then can you please remove from this file because its looking weired :v: